rskj-core/src/main/java/co/rsk/trie/SharedPathSerializer.java

```diff
@@ -0,0 +1,153 @@
+        this.lshared = this.sharedPath.length();
+    }
+
+
+    public int serializedLength() {
+        if (!isPresent()) {
+            return 0;
+        }
+
+        return lsharedSize() + PathEncoder.calculateEncodedLength(lshared);
+    }
+
+    public boolean isPresent() {
+        return lshared > 0;
+    }
+
+    public void serializeInto(ByteBuffer buffer) {
+      serializeInto(this.sharedPath,buffer);
+    }
+
+    public static void serializeInto(TrieKeySlice sharedPath,ByteBuffer buffer) {
+        if (!isPresent(sharedPath)) {
+            return;
+        }
+        int lshared = sharedPath.length();
+        if (1 <= lshared && lshared <= 32) {
+            // first byte in [0..31]
+            buffer.put((byte) (lshared - 1));
+        } else if (160 <= lshared && lshared <= 382) {
+            // first byte in [32..254]
+            buffer.put((byte) (lshared - 128));
+        } else {
+            buffer.put((byte) 255);
+            buffer.put(new VarInt(lshared).encode());
+        }
+
+        buffer.put(sharedPath.encode());
+    }
+
+    // Returns the size of the path prefix when path needs encoding.
+    // The prefix indicates the length of the shared path in bits.
+    // TO DO: Rename to something like getEncodedPathBitlengthSize()
+    private static int lsharedSize(TrieKeySlice sharedPath) {
+        if (!isPresent(sharedPath)) {
+            return 0;
+        }
+        int lshared = sharedPath.length();
+        if (1 <= lshared && lshared <= 32) {
+            return 1;
+        }
+
+        if (160 <= lshared && lshared <= 382) {
+            return 1;
+        }
+
+        return 1 + VarInt.sizeOf(lshared);
+    }
+
+    private int lsharedSize() {
+        return lsharedSize(this.sharedPath);
+    }
+
+    public static int getPathBitsLength(ByteBuffer message) {
+        int lshared; // this is a bit length
+
+        // upgrade to int so we can compare positive values
+        int lsharedFirstByte = Byte.toUnsignedInt(message.get());
+        if (0 <= lsharedFirstByte && lsharedFirstByte <= 31) {
+            // lshared in [1..32]
+            lshared = lsharedFirstByte + 1;
+        } else if (32 <= lsharedFirstByte && lsharedFirstByte <= 254) {
+            // lshared in [160..382]
+            lshared = lsharedFirstByte + 128;
+        } else {
+            lshared = (int) readVarInt(message).value;
+        }
+        return lshared;
+    }
+    public static TrieKeySlice deserialize(ByteBuffer message, boolean sharedPrefixPresent) {
+        if (!sharedPrefixPresent) {
+            return TrieKeySlice.empty();
+        }
+
+        int lshared = getPathBitsLength(message);
+
+        int lencoded = PathEncoder.calculateEncodedLength(lshared);
+        byte[] encodedKey = new byte[lencoded];
+        message.get(encodedKey);
+        return TrieKeySlice.fromEncoded(encodedKey, 0, lshared, lencoded);
+    }
+
+    private static VarInt readVarInt(ByteBuffer message) {
```

Comment:
Duplicated from Trie? Maybe better to extract into a `VarInt` static class or similar, in name of the refactor for clarity.

Reply:
ok. I'll do it

Reply:
Are we going to extract this method or it will be added in a future PR?